pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -552,11 +552,11 @@ func (p *Planner) reconcile(controlPlane *rkev1.RKEControlPlane, secret plan.Sec
 		} else if !equality.Semantic.DeepEqual(entry.Plan.Plan, plan) {
 			outOfSync = append(outOfSync, entry.Machine.Name)
 			// Conditions
-			// 1. If plan is not in sync then there is no harm in updating it to something else because
-			//    the node will have already been considered unavailable.
+			// 1. If the node is already draining then the plan is out of sync.  There is no harm in updating it if
+			// the node is currently drained.
 			// 2. concurrency == 0 which means infinite concurrency.
 			// 3. unavailable < concurrency meaning we have capacity to make something unavailable
-			if isUnavailable(entry) || concurrency == 0 || unavailable < concurrency {
+			if isInDrain(entry.Machine) || concurrency == 0 || unavailable < concurrency {
 				if !isUnavailable(entry) {
 					unavailable++
```

Comment:
In the case that we enter the clause because concurrency == 0 or unavailable < concurrency,  the current entry is available, so should we decrease the count of unavailable? 
Or do I misunderstand the logic here? 

```suggestion
				if !isUnavailable(entry) {
					unavailable--
```

Reply:
That is for when the plan is out of sync but the node is already draining.  In that case it was already added to our unavailable count we just want to refresh the plan.